packages/flutter/lib/src/cupertino/list_tile.dart

```diff
@@ -0,0 +1,412 @@
+  /// not be used to toggle a trailing [CupertinoSwitch] and similar usecases
+  /// because when tile is tapped, it switches the background color and remains
+  /// changed. This is according to iOS behaviour.
+  ///
+  /// The [backgroundColor] provides a custom background color for the tile in
+  /// a state before tapped. By default, it matches the theme's background color
+  /// which is by default a [CupertinoColors.systemBackground].
+  ///
+  /// The [backgroundColorActivated] provides a custom background color for the
+  /// tile after it was tapped. By default, it matches the theme's background
+  /// color which is by default a [CupertinoColors.systemGrey4].
+  ///
+  /// The [padding] parameter sets the padding of the content inside the tile.
+  /// It defaults to a value that matches the iOS look, depending on a type of
+  /// [CupertinoListTile]. For native look, it should not be provided.
+  ///
+  /// The [leadingSize] constrains the width and height of the leading widget.
+  /// By default, it is set to a value that matches the iOS look, depending on a
+  /// type of [CupertinoListTile]. For native look, it should not be provided.
+  ///
+  /// The [leadingToTitle] specifies the horizontal space between [leading] and
+  /// [title] widgets. By default, it is set to a value that matched the iOS
+  /// look, depending on a type of [CupertinoListTile]. For native look, it
+  /// should not be provided.
+  const CupertinoListTile.notched({
+    Key? key,
+    required this.title,
+    this.subtitle,
+    this.additionalInfo,
+    this.leading,
+    this.trailing,
+    this.onTap,
+    this.backgroundColor,
+    this.backgroundColorActivated,
+    this.padding,
+    this.leadingSize = _kNotchedLeadingSize,
+    this.leadingToTitle = _kNotchedLeadingToTitle,
+  }) : _type = _CupertinoListTileType.notched,
+       super(key: key);
+
+  final _CupertinoListTileType _type;
+
+  /// A [title] is used to convey the central information. Usually a [Text].
+  final Widget title;
+
+  /// A [subtitle] is used to display additional information. It is located
+  /// below [title]. Usually a [Text] widget.
+  final Widget? subtitle;
+
+  /// Similar to [subtitle], an [additionalInfo] is used to display additional
+  /// information. However, instead of being displayed below [title], it is
+  /// displayed on the right, before [trailing]. Usually a [Text] widget.
+  final Widget? additionalInfo;
+
+  /// A widget displayed at the start of the [CupertinoListTile]. This is
+  /// typically an `Icon` or an `Image`.
+  final Widget? leading;
+
+  /// A widget displayed at the end of the [CupertinoListTile]. This is usually
+  /// a right chevron icon (e.g. `CupertinoListTileChevron`), or an `Icon`.
+  final Widget? trailing;
+
+  /// The [onTap] function is called when a user taps on [CupertinoListTile]. If
+  /// left `null`, the [CupertinoListTile] will not show any visual information
+  /// when tapped.
+  final Future<void> Function()? onTap;
```

Comment:
**question**: Why is this declared as `final Future<void> Function()? onTap;` rather than `final void Function()? onTap;` (or, in an equivalent form, `final VoidCallback? onTap;`)?

Reply:
You are right. I think I have had an usecase where this needed to be a `Future<void>` but it makes sense to turn this into a `VoidCallback`.

Reply:
This should not be a `VoidCallback` for the following reason: pressing a "list tile" on iOS that navigates to a new route, highlights that list tile. Once the route is poped, the list tile is still highlighted in remains to be so, until route is completely popped. If this function completes instantly, given current implementation (simple yet effective) the list tile become not highlighted instantly.

It would make sense to turn this into a `FutureOr<void>` function, however AFAIR this is against Flutter coding standards. I will however still convert this to a `FutureOr<void> Function()? onTap`.